src/app/modules/user/user.controller.ts

```diff
@@ -206,3 +214,51 @@ export const handleFetchUser: ControllerHandler = async (req, res) => {
       return res.status(statusCode).json({ message: errorMessage })
     })
 }
+
+export const _handleUpdateUserLastSeenFeatureUpdateDate: ControllerHandler<
+  unknown,
+  string,
+  UpdateUserLastFeatureUpdateDateDto
+> = async (req, res) => {
+  const { latestLastSeenFeatureUpdateDate } = req.body
+  const sessionUserId = getUserIdFromSession(req.session)
+
+  if (!sessionUserId) {
+    return res.status(StatusCodes.UNAUTHORIZED).json(UNAUTHORIZED_USER_MESSAGE)
+  }
+
+  return updateUserLastSeenFeatureUpdateDate(
+    sessionUserId,
+    latestLastSeenFeatureUpdateDate,
+  )
+    .map(() => {
+      return res
+        .status(StatusCodes.OK)
+        .json(UPDATE_LAST_SEEN_FEATURE_DATE.SUCCESS)
```

Comment:
do you think we should return the updated user back to the client instead? So the user can be updated on the client using data from the server.

Reply:
Now that I think about it, I think we would need to because the User object is cached on the client side because of `useQuery`. Hence if there is an update being done to the user and it's not returned, the last seen feature update date attribute on the client side will be stale for a period of time and this might cause incorrect behaviour when deciding whether or not to notify the user on the feature updates.